tests/libunwind/enc/CMakeLists.txt

```diff
@@ -52,6 +52,30 @@ function(add_libunwind_test_enc NAME TESTFILE)
     target_link_libraries(libunwindtest-${NAME}_enc libunwindtest-support)
     target_link_libraries(libunwindtest-${NAME}_enc -Wl,--undefined=pthread_mutex_lock)
     target_link_libraries(libunwindtest-${NAME}_enc -Wl,--undefined=_U_dyn_info_list_addr)
+
+    maybe_build_using_clangw(libunwindtest-${NAME}_enc)
+
+    # maybe_build_using_clangw populates variables in its parent scope (ie current scope)
```

Comment:
Is this done in other tests? I can look more into this when I have more time, it just seems a bit cumbersome.

Reply:
Hmm, I haven't seen this pattern before, but I think it's because `maybe_build_using_clangw` is being invoked in a function which is called in a loop in the normal part of this CMakeLists.txt file. And according to the cmake documentation, functions create new scopes (which makes sense). Invoking `maybe_build_using_clangw` here will populate those `CMAKE_*` variables in this function's scope, but it won't set those variables in the foreach loop that is calling this function.

If the `add_libunwind_test_enc` cmake function was removed and then all this code was embedded directly in the loop itself, the propagation code below probably wouldn't be necessary. That's just my guess though!

Honestly the `maybe_build_using_clangw` function is really awkward but I don't know of a better way to do what it does (uses a second compile/link toolchain to build separate executables within the same cmake build). Maybe someone can think of a better solution :)

Reply:
oh, I get it now. I didn't realize that this was being called inside a function. Thanks!